Tests/UnitTests/Paywalls/Events/PaywallEventSerializerTests.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PaywallEventSerializerTests.swift
+//
+//  Created by Nacho Soto on 9/5/23.
+
+import Foundation
+import Nimble
+@testable import RevenueCat
+
+class PaywallEventSerializerTests: TestCase {
+
+    func testEncodeView() throws {
+        let data: PaywallEvent.Data = .init(
+            offeringIdentifier: "offering",
+            paywallRevision: 5,
+            sessionID: .init(),
+            displayMode: .condensedFooter,
+            requestedLocaleIdentifier: "es_ES",
+            displayedLocaleIdentifier: "es",
+            darkMode: Bool.random()
+        )
+        let event: PaywallEvent = .view(data, Date().removingMilliseconds)
+
+        let encoded = try PaywallEventSerializer.encode(event)
+
+        let test = String(data: try JSONEncoder.prettyPrinted.encode(event), encoding: .utf8)!
+        Logger.error(test)
+
+        let decoded: PaywallEvent = try JSONDecoder.default.decode(jsonData: encoded.asData)
+
+        expect(encoded.numberOfLines) == 1
+        expect(decoded) == event
+    }
+
```

Comment:
Gotta test the other events, just waiting for confirmation if we're going to be sending the whole data again on those or just the session ID /cc @joshdholtz